course/views.py

```diff
@@ -106,8 +106,22 @@ def get(self, request, *args, **kwargs):
         elif lti_msg:
             messages.info(request, html.escape(lti_msg))
 
-        return super().get(request, *args, **kwargs)
+        later_instance = None
+
+        if self.instance.is_past:
+            try:
+                later_instance = (
+                    CourseInstance.objects
+                        .get_visible(request.user)
```

Comment:
Since `request` is an argument of the function, it can be accessed straightaway, without using the `self`. Works both ways though, with the `self` as well.